### PR TITLE
Build specialization functionality

### PR DESCRIPTION
**Pull request by Maha-Magdy**
Hello @eapenzacharias ☺,
I this pull request I have done 
## Create the routes and controller for the specialization 

- [x] Create routes that needed for the specialization 

- [x] Create the controller with its methods:
   - [x] Create a new specialization 
   - [x] Index all specializations
   - [x] Show one specific specialization 
   - [x] Edit a specific specialization 
   - [x] Delete specific specialization 

- [x] Use TTD

### CODE REVIEW

File: app/controllers/concerns/response.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module Response
```
Comment (eapenzacharias) [inline app/controllers/concerns/response.rb:1]: This was a good approach to keep your controller dry 👍

File: spec/integration/specialization_spec.rb [hunk 2]
```diff
@@ -0,0 +1,128 @@
+require 'swagger_helper'
+
+describe 'Specializations API' do
```
Comment (eapenzacharias) [inline spec/integration/specialization_spec.rb:3]: We have to include authorization headers for tests that can be done at a later stage

**Review by eapenzacharias — APPROVED** [review]
All the necessary requirements are met and we can finalize the testing at a later stage.👍

Congrats on the good work! 🎉

File: app/controllers/concerns/response.rb (hunk 1 above)
Comment (Maha-Magdy) [reply app/controllers/concerns/response.rb:1]: Thanks, @eapenzacharias 😊✌

**Pull request merged**